x-pack/filebeat/input/awss3/config.go

```diff
@@ -151,6 +152,36 @@ func (rc *readerConfig) Validate() error {
 	return nil
 }
 
+type scriptConfig struct {
+	Source            string                 `config:"source"`                               // Inline script to execute.
+	File              string                 `config:"file"`                                 // Source file.
+	Files             []string               `config:"files"`                                // Multiple source files.
```

Comment:
Maybe document the order of script source selection; Source>File>Files. Also in input-aws-s3.asciidoc.

Reply:
There is a comment in the function about only one being allowed at the same time (so no preference, if there are more than one an error is shown). But this was lacking in the docs so I added it there, also.